ethcore/src/client/test_client.rs

```diff
@@ -351,17 +370,116 @@ impl TestBlockChainClient {
 	pub fn set_history(&self, h: Option<u64>) {
 		*self.history.write() = h;
 	}
+
+    /// Create a signed transaction destined for `address`
+    /// `data` is an ethabi encoded function call
+    pub fn contract_call_tx(&self, id: BlockId, address: Address, data: Bytes) -> SignedTransaction {
+		let from = Address::default();
+		Transaction {
+			nonce: self.nonce(&from, id).unwrap_or_else(|| self.spec.engine.account_start_nonce(0)),
+			action: Action::Call(address),
+			gas: U256::from(50_000_000),
+			gas_price: U256::default(),
+			value: U256::default(),
+			data: data,
+		}.fake_sign(from)
+    }
+
+    /// Create a signed transaction to deploy a contract encoded in the `data` parameter 
+    pub fn contract_create_tx(&self, id: BlockId, address: Address, data: Bytes) -> SignedTransaction {
+		Transaction {
+			nonce: self.nonce(&address, id).unwrap_or_else(|| self.spec.engine.account_start_nonce(0)),
+			action: Action::Create,
+			gas: U256::from(50_000_000),
+			gas_price: U256::default(),
+			value: U256::default(),
+			data: data,
+		}.fake_sign(address)
+    }
+
+    /// Evaluate a transaction on a local EVM Interpreter
+	fn do_virtual_call(&self, env_info: &EnvInfo, state: &mut State<StateDB>, t: &SignedTransaction, analytics: CallAnalytics) -> Result<Executed, CallError> {
+		fn call<V, T>(
+			state: &mut State<StateDB>,
+			env_info: &EnvInfo,
+			machine: &::machine::EthereumMachine,
+			state_diff: bool,
+			transaction: &SignedTransaction,
+			options: TransactOptions<T, V>,
+		) -> Result<Executed<T::Output, V::Output>, CallError> where
+			T: trace::Tracer,
+			V: trace::VMTracer,
+		{
+			let options = options
+				.dont_check_nonce()
+				.save_output_from_contract();
+			let original_state = if state_diff { Some(state.clone()) } else { None };
+
+			let mut ret = Executive::new(state, env_info, machine).transact_virtual(transaction, options)?;
+
+			if let Some(original) = original_state {
+				ret.state_diff = Some(state.diff_from(original).map_err(ExecutionError::from)?);
+			}
+			Ok(ret)
+		}
+
+		let state_diff = analytics.state_diffing;
+		let machine = self.spec.engine.machine();
+
+		match (analytics.vm_tracing,) {
+			(true,) => call(state, env_info, machine, state_diff, t, TransactOptions::with_vm_tracing()),
+			(false,) => call(state, env_info, machine, state_diff, t, TransactOptions::with_no_tracing()),
+		}
+	}
+
+    /// Convenience method to retrieve current state from a given state_db
+    pub fn get_in_memory_state(&self) -> Result<State<StateDB>, EvmTestError> {
```

Comment:
Avoid `get` in function names use `in_memory_state` is it's a getter or `new_/create_` if you are creating something.